Objects/listobject.c

```diff
@@ -478,18 +478,21 @@ list_length(PyObject *a)
 static int
 list_contains(PyObject *aa, PyObject *el)
 {
-    PyListObject *a = (PyListObject *)aa;
-    PyObject *item;
-    Py_ssize_t i;
-    int cmp;
 
-    for (i = 0, cmp = 0 ; cmp == 0 && i < Py_SIZE(a); ++i) {
-        item = PyList_GET_ITEM(a, i);
-        Py_INCREF(item);
-        cmp = PyObject_RichCompareBool(item, el, Py_EQ);
+    for (Py_ssize_t i = 0; ; i++) {
+        PyObject *item = PyList_GetItemRef(aa, i);
+        if (item == NULL && PyErr_ExceptionMatches(PyExc_IndexError)) {
```

Comment:
I think we should have a `static` function that is like `PyList_GetItemRef`, but does not set exceptions and use that here.